MockKit/MockPumpManager.swift

```diff
@@ -402,14 +426,22 @@ public final class MockPumpManager: TestingPumpManager {
 
         logDeviceCommunication("enactBolus(\(units), \(startDate))")
 
-        if state.bolusEnactmentShouldError {
+        if state.bolusEnactmentShouldError || state.pumpBatteryChargeRemaining == 0 {
             let error = PumpManagerError.communication(MockPumpManagerError.communicationFailure)
             logDeviceComms(.error, message: "Bolus failed with error \(error)")
             completion(.failure(error))
         } else if state.deliveryCommandsShouldTriggerUncertainDelivery {
             state.deliveryIsUncertain = true
             logDeviceComms(.error, message: "Uncertain delivery for bolus")
             completion(.failure(PumpManagerError.uncertainDelivery))
+        } else if state.occlusionDetected || state.pumpErrorDetected {
+            let error = PumpManagerError.deviceState(MockPumpManagerError.pumpError)
+            logDeviceComms(.error, message: "Bolus failed because the pump is in an error state")
+            completion(.failure(error))
+        } else if state.reservoirUnitsRemaining <= 0 {
```

Comment:
Should there be a check for `case .suspended = state.suspendState`?

Reply:
there is a bit further down.